module/Database/src/Service/Mailman.php

```diff
@@ -210,6 +211,15 @@ private function syncMembershipSingle(
 
         $verifyTime = (new DateTime())->sub(new DateInterval('P1D'));
 
+        $isMailmanList = $dbList->hasMailmanList();
+        if ($isMailmanList) {
+            $listId = $dbList->getMailmanList()->getMailmanId();
+            $knownMembers = $this->getMailmanListSubscriberEmails($listId);
+        } else {
+            // This is to satisfy psalm, observe that it is not needed
+            $knownMembers = [];
+        }
```

Comment:
Psalm wants you to declare it before the `if` statement; however, it is a non-issue for this review.